pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,9 +91,56 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	contents, err := p.createSecrets(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("creating secrets object: %v", err)
+	}
+
+	err = p.providerKubectlClient.ApplyKubeSpecFromBytes(ctx, cluster, contents)
+	if err != nil {
+		return fmt.Errorf("applying secrets object: %v", err)
+	}
 	return nil
 }
 
+func (p *cloudstackProvider) createSecrets(ctx context.Context, cluster *types.Cluster) ([]byte, error) {
+	if err := p.providerKubectlClient.CreateNamespaceIfNotExists(ctx, cluster.KubeconfigFile, constants.EksaSystemNamespace); err != nil {
+		return nil, err
+	}
+
+	secrets := [][]byte{}
+	for _, profile := range p.execConfig.Profiles {
+		secret := createSecret(profile)
+		if bytes, err := yaml.Marshal(secret); err == nil {
+			secrets = append(secrets, bytes)
+		} else {
+			return nil, fmt.Errorf("marshalling secret: %v: %v", secret, err)
+		}
+	}
+	return templater.AppendYamlResources(secrets...), nil
+}
+
+func createSecret(profile decoder.CloudStackProfileConfig) *corev1.Secret {
+	immutable := true
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.Version,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: constants.EksaSystemNamespace,
+			Name:      profile.Name,
+		},
+		Immutable: &immutable,
```

Comment:
I don't think we want to set this attribute - users should be able to modify/rotate the values here, but that's impossible with this attribute set as far as I can tell

Reply:
Removed Immutable: true from secret creation